--- a/target/linux/mediatek/image/filogic.mk
+++ b/target/linux/mediatek/image/filogic.mk
@@ -1,3 +1,21 @@
+# Netcore N30 Pro
+define Device/netcore_n30pro
+  DEVICE_VENDOR := Netcore
+  DEVICE_MODEL := N30 Pro
+  DEVICE_DTS := mt7981b-netcore-n30pro
+  DEVICE_DTS_DIR := ../dts
+  SUPPORTED_DEVICES += netcore,n30pro
+  DEVICE_PACKAGES := kmod-mt7915-firmware kmod-mt7981-firmware \
+	kmod-usb3 kmod-usb-xhci-mtk kmod-mt7531
+  IMAGE/sysupgrade.bin := append-kernel | pad-to 128k | append-rootfs | \
+	pad-rootfs | append-metadata
+  IMAGE/initramfs.itb := append-kernel | pad-to 128k | append-rootfs | \
+	pad-rootfs | append-metadata
+endef
+TARGET_DEVICES += netcore_n30pro
+
 define Device/mediatek_mt7981_evb
   DEVICE_VENDOR := MediaTek
   DEVICE_MODEL := MT7981 EVB
